module/icp/core/kcf_sched.c

```diff
@@ -1740,8 +1740,13 @@ kcf_last_req(void *last_req_arg, int status)
 		ct = (crypto_dual_data_t *)dcrops->dop_ciphertext;
 		break;
 	}
-	default:
-		break;
+	default: {
+		crypto_call_req_t *cr = &last_req->kr_callreq;
+
+		(*(cr->cr_callback_func))(cr->cr_callback_arg, status);
+		kmem_free(last_req, sizeof (kcf_dual_req_t));
+		return;
```

Comment:
Running the callback in this case and freeing the memory seems like a sane way to gracefully handle this impossible case but I'd like to get @tcaputi take on this.  Should this really be fatal since it's in the icp code?